clients/@grouparoo/web/README.md

```diff
@@ -0,0 +1,121 @@
+- **path** (optional): The route to post events too. Default: `/api/v1/track`
+- **anonymousId** (optional): An ID you would like to use for the `anonymousId` for this client. By default, we will generate a unique UUID for this browser and store it in a cookie for future use.
+- **errorHandlers** (optional): An array of functions to pass errors too. This is helpful if you want to display tracking errors to the user, or log them with a tool like New Relic.
+- **logging** (optional): Should tracking events be logged to browser console? Default: `false`
+- **cookie** (optional): What is the name of the cookie to store the `anonymousId`? Default: `grouparoo-anonymous-id`
+- **cookieTTL** (optional): How long should the tracking cookie exist for? Default: 90 days.
+  - This setting will effect your session duration for `new_session` events triggered the `page` higher-order event.
+
+## Creating Events
+
+**Track**
+
+To create an event, call the `track` method on the client with a `type` and any additional data you want to store about the event:
+
+```ts
+// track an arbitrary event
+// API: client.track(event-name, data?)
+client.track("button-click", { page: "/home", button: "learn-more" });
+```
+
+This will send the event, time, and associated user information (`anonymousId` and `userId` if set) to your Grouparoo server.
+
+**Identify**
+
+A special event called `identify` should be called whenever you know a way to de-anonymize this user. This usually corresponds with a `sign-in` event. The `identify` event requires your application's unique id for this user set as the `userId` argument.
+
+```ts
+// The user successfully signed in as user #123
+// API: client.identify(userId, data?)
+client.identify(123, { firstName: "Mario", lastName: "Mario" });
+client.track("sign-in", { page: "/sign-in" });
+```
+
+From this point onward, all events tracked will be linked to user #123. Grouparoo will also associate all previous events from this session (with the same `anonymousId`) to user #123.
+
+The second argument to `identify`, `data`, will _not_ override profile data that Grouparoo has obtained from your Sources. However, it may be useful to record related identifying information about the profile at the time of `identify`. This information will be stored as data on the Event, and able to be later utilized within Grouparoo.
+
+**Reset**
+
+The `reset` method will delete all data stored, and reset the `userId` and `anonymousId`. Use this method when a user logs out.
+
+```ts
+// The user has logged out
+// API: client.reset(newAnonymousId?)
+client.reset();
+```
+
+## Higher-Order Events
+
+This library also comes with a few higher-order events geared towards making it easy to track the performance of your websites.
+
+**Page**
+
+The `page` method will create a track event pre-filled with some useful data about the page. The `page` event will also create a `new_session` event for the first time that an `anonymousId` has viewed any page.
```

Comment:
I wonder if `new_session` should be a property on the event and not a whole other event.

Reply:
I also wonder if segment or others has a "sessionId" on every event allowing aggregating by session in an interesting way.

Reply:
AFAIK, this isn't something segment does.  This pattern I'm taking from Google Analytics... but it might be overkill.  We count unique anonymousId's just as easily as we can count `new_session` events.  We can also easily find the first events for any given anonymousId with some property.  I think I'll remove this.